yellowbrick/classifier/prcurve.py

```diff
@@ -0,0 +1,472 @@
+            return self._draw_binary()
+        return self._draw_multiclass()
+
+    def _draw_binary(self):
+        """
+        Draw the precision-recall curves in the binary case
+        """
+        self._draw_pr_curve(self.recall_, self.precision_, label="binary PR curve")
+        self._draw_ap_score(self.score_)
+
+
+    def _draw_multiclass(self):
+        """
+        Draw the precision-recall curves in the multiclass case
+        """
+        # TODO: handle colors better with a mapping and user input
+        if self.per_class:
+            for cls in self.classes_:
+                precision = self.precision_[cls]
+                recall = self.recall_[cls]
+
+                label = "PR for class {} (area={:0.2f})".format(cls, self.score_[cls])
+                self._draw_pr_curve(recall, precision, label=label)
+
+        if self.micro:
+            precision = self.precision_[MICRO]
+            recall = self.recall_[MICRO]
+            self._draw_pr_curve(recall, precision)
+
+        self._draw_ap_score(self.score_[MICRO])
+
+    def _draw_pr_curve(self, recall, precision, label=None):
+        """
+        Helper function to draw a precision-recall curve with specified settings
+        """
+        self.ax.step(
+            recall, precision, alpha=self.line_opacity, where='post', label=label
+        )
+        if self.fill_area:
+            self.ax.fill_between(recall, precision, step='post', alpha=self.fill_opacity)
+
+    def _draw_ap_score(self, score, label=None):
+        """
+        Helper function to draw the AP score annotation
+        """
+        label = label or "AP={:0.2f}".format(score)
+        if self.ap_score:
+            self.ax.axhline(
+                y=score, color="r", ls="--", label=label
+            )
+
+    def finalize(self):
+        """
+        Finalize the figure by adding titles, labels, and limits.
+        """
+        self.set_title("Precision-Recall Curve for {}".format(self.name))
+        self.ax.legend(loc='lower left', frameon=True)
+
+        self.ax.set_xlim([0.0, 1.0])
+        self.ax.set_ylim([0.0, 1.0])
+
+        self.ax.set_ylabel("Precision")
+        self.ax.set_xlabel("Recall")
+
+    def _get_y_scores(self, X):
+        """
+        The ``precision_recall_curve`` metric requires target scores that
+        can either be the probability estimates of the positive class,
+        confidence values, or non-thresholded measures of decisions (as
+        returned by a "decision function").
+
+        .. todo:: refactor shared method with ROCAUC
+        .. todo:: ensure only probabilities from positive case are returned (binary)
+        """
+        # Resolution order of scoring functions
+        attrs = (
+            'decision_function',
+            'predict_proba',
+        )
+
+        # Return the first resolved function
+        for attr in attrs:
+            try:
+                method = getattr(self.estimator, attr, None)
```

Comment:
It is - but it's preferable to use try/except here since the implementation of [`hasattr`](https://docs.python.org/3/library/functions.html#hasattr) is: 

```python
def hasattr(obj, attr):
    try:
        getattr(obj, attr)
    except AttributeError:
        return False 
    return True
```

Generally speaking, Python prefers [EAFP](https://docs.python.org/3/glossary.html?highlight=eafp#term-eafp) (easier to ask for forgiveness than permission) rather than LBYL (look before you leap). Following this principle, code will generally be faster (in this case by preventing 2 calls to `getattr`).